hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyInputStream.java

```diff
@@ -351,14 +351,14 @@ public boolean seekToNewSource(long targetPos) throws IOException {
   }
 
   @Override
-  public int available() throws IOException {
+  public synchronized int available() throws IOException {
```

Comment:
This seems to return remaining value, if multiple people are operating on this stream, with synchronised also this can give stale values.
Just out of curiosity: Any pointers in which use case having this synchronised will help

Reply:
BTW, available() in S3AInputStream and HDFS's DFSInputStream are both synchronized.

Reply:
> if multiple people are operating on this stream, with synchronised also this can give stale values

Correct, that may be a design choice of `InputStream` in general.

However, synchronization helps not only with concurrent usage, but also sequential usage in multiple threads, as it provides visibility guarantees.

I don't have a use case for making this specific method `synchronized`, but nor do I for the other methods that are already synced.  It's just a matter of consistency.